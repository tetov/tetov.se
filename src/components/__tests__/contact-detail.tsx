import React from "react";
import renderer from "react-test-renderer";
import ContactDetail from "src/components/contact-detail";

describe("ContactDetail", () => {
  it("renders correctly with icon & url", () => {
    const tree = renderer
      .create(
        <ContactDetail
          text="MySpace"
          url="https://myspace.tk/tom"
          label="name"
          useIcon
          className="where-is-he-now where-did-he-go"
          hcard="u-url"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly without icon but with url", () => {
    const tree = renderer
      .create(
        <ContactDetail
          text="MySpace"
          url="https://myspace.tk/tom"
          className="where-is-he-now where-did-he-go"
          hcard="u-url"
          label="name"
          useIcon={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with icon but without url", () => {
    const tree = renderer
      .create(
        <ContactDetail
          text="MySpace"
          className="where-is-he-now where-did-he-go"
          hcard="u-url"
          useIcon
          label="name"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly without icon and without url", () => {
    const tree = renderer
      .create(
        <ContactDetail
          text="MySpace"
          className="where-is-he-now where-did-he-go"
          hcard="u-url"
          label="name"
          useIcon={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
