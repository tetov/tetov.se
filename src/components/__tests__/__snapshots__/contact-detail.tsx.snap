// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetail renders correctly with icon & url 1`] = `
<div
  className="flex group p-2"
>
  <a
    className="u-url where-is-he-now where-did-he-go"
    href="https://myspace.tk/tom"
    rel="me external"
  >
    <div
      className="motion-safe:group-hover:animate-pulse inline-block"
    >
      <svg
        aria-hidden={true}
        className="inline"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zM7 12a5 5 0 0 0 10 0h-2a3 3 0 0 1-6 0H7z"
          />
        </g>
      </svg>
    </div>
    <span
      className="pl-1"
    >
      MySpace
    </span>
  </a>
</div>
`;

exports[`ContactDetail renders correctly with icon but without url 1`] = `
<div
  className="flex group p-2"
>
  <span
    className="u-url where-is-he-now where-did-he-go"
  >
    <div
      className="motion-safe:group-hover:animate-pulse inline-block"
    >
      <svg
        aria-hidden={true}
        className="inline"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zM7 12a5 5 0 0 0 10 0h-2a3 3 0 0 1-6 0H7z"
          />
        </g>
      </svg>
    </div>
    <span
      className="pl-1"
    >
      MySpace
    </span>
  </span>
</div>
`;

exports[`ContactDetail renders correctly without icon and without url 1`] = `
<div
  className="flex group p-2"
>
  <span
    className="u-url where-is-he-now where-did-he-go"
  >
    <span
      className=""
    >
      MySpace
    </span>
  </span>
</div>
`;

exports[`ContactDetail renders correctly without icon but with url 1`] = `
<div
  className="flex group p-2"
>
  <a
    className="u-url where-is-he-now where-did-he-go"
    href="https://myspace.tk/tom"
    rel="me external"
  >
    <span
      className=""
    >
      MySpace
    </span>
  </a>
</div>
`;
